crates/ra_parser/src/grammar/expressions.rs

```diff
@@ -305,9 +311,13 @@ fn lhs(p: &mut Parser, r: Restrictions) -> Option<(CompletedMarker, BlockLike)>
         }
         // test unary_expr
         // fn foo() {
-        //     **&1;
+        //     **&1 + 1;
         //     !!true;
         //     --1;
+        //     *&1 as u64;
+        //     *x(1);
+        //     &x[1];
+        //     -1..2;
```

Comment:
Let's move these additional cases to an out-of-line test somewhere in test-data/parser/ok. The purpose of the inline tests is not to exhaustively check everything, but to explain what the code immediately after them does, and, for that purposes, simple cases are more useful.